server.go

```diff
@@ -44,8 +44,15 @@ import (
 // coarseTimeProvider).
 var coarseTime *coarseTimeProvider
 
-func init() {
-	coarseTime = newcoarseTimeProvider(25 * time.Millisecond)
+// coarseTimeOnce is used to ensure we create a single coarseTime instance.
+var coarseTimeOnce sync.Once
+
+// initCoarseTime is used to ensure the global coarseTime is initialized.
+func initCoarseTime() *coarseTimeProvider {
```

Comment:
I don't think it needs to return the pointer if it's setting the global var - I guess this is a refactoring artefact?